commit d589b8d9a4c83f9021131e186f2d39187ba95a2b
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Fri Mar 13 12:43:36 2015 +0000

    Protect __release_resource against resources without parents
    
    Without this patch, removing a device tree overlay can crash here.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/kernel/resource.c b/kernel/resource.c
index 9b5f04404152..f8a9af6e6b91 100644
--- a/kernel/resource.c
+++ b/kernel/resource.c
@@ -246,6 +246,12 @@ static int __release_resource(struct resource *old, bool release_child)
 {
 	struct resource *tmp, **p, *chd;
 
+	if (!old->parent) {
+		WARN(old->sibling, "sibling but no parent");
+		if (old->sibling)
+			return -EINVAL;
+		return 0;
+	}
 	p = &old->parent->child;
 	for (;;) {
 		tmp = *p;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1004_d589b8d9a4c83f9021131e186f2d39187ba95a2b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1004_d589b8d9a4c83f9021131e186f2d39187ba95a2b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
